### PR TITLE
TokensAnalyzer - isBinaryOperator fix for intersection

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/6317

### CODE REVIEW

File: tests/Tokenizer/TokensAnalyzerTest.php [hunk 1]
```diff
@@ -1791,6 +1791,32 @@ public static function provideIsBinaryOperator80Cases(): iterable
...
+                static::assertFalse($tokensAnalyzer->isUnaryPredecessorOperator($index));
+            }
+        }
+    }
+
+    public static function provideIsBinaryOperator81Cases(): iterable
+    {
+        yield 'intersection is not a binary operator' => [
+            [6 => false],
+            '<?php function test(A&B $var): void {}',
```
Comment (kubawerlos) [inline tests/Tokenizer/TokensAnalyzerTest.php:1816]: Can we have here very similar code like `test(A&B)` and `function test($var = A&B) {}` as in both `&` is bitwise "and"?

Comment (SpacePossum) [reply tests/Tokenizer/TokensAnalyzerTest.php:1829]: Added more tests, please see last commit

**Comment by SpacePossum** [comment]
current code (so not introduced here);
consider `&` in `<?php function replace(?int &$count){}` TokensAnalyzer considers this a `isUnaryPredecessorOperator` 

for example
```php
            [
                [3 => true],
                '<?php function &foo() {}',
            ],

            [
                [5 => true, 11 => true, 17 => true],
                '<?php function foo(&$a, array &$b, Bar &$c) {}',
            ],
```

I think this is wrong as these aren't operators, just references, requires bunch of work

**Comment by kubawerlos** [comment]
Yeah, this is a mess, there is `ReferenceAnalyzer` which should detect is as well.

**Pull request closed without merging**